src/YearMonth.php

```diff
@@ -44,17 +47,25 @@ private function __construct(int $year, int $month)
     /**
      * Obtains an instance of `YearMonth` from a year and month.
      *
-     * @param int $year  The year, from MIN_YEAR to MAX_YEAR.
-     * @param int $month The month-of-year, from 1 (January) to 12 (December).
+     * @param int $year The year, from MIN_YEAR to MAX_YEAR.
      *
      * @throws DateTimeException
      */
-    public static function of(int $year, int $month): YearMonth
+    public static function of(int $year, int|Month $month): YearMonth
     {
         Field\Year::check($year);
-        Field\MonthOfYear::check($month);
 
-        return new YearMonth($year, $month);
+        if (is_int($month)) {
+            // usually we don't use trigger_error() for deprecations, but we can't rely on @deprecated for a parameter type change;
+            // maybe we should revisit using trigger_error() unconditionally for deprecations in the future.
+            trigger_error('Passing an integer to YearMonth::of() second argument is deprecated, pass a Month instance instead.', E_USER_DEPRECATED);
+
+            Field\MonthOfYear::check($month);
+
+            $month = Month::from($month);
```

Comment:
Shouldn't you return `new YearMonth($year, $month);` straight away here? Otherwise you're instantiating an enum for nothing!